src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -124,61 +130,63 @@ public ImmutableArray<T> Batch()
             return sourceBuilder.ToImmutableAndFree();
         }
 
-        public Builder ToBuilder(int? extraCapacity = null)
+        public Builder ToBuilder()
         {
             Debug.Assert(!this.IsFaulted);
-            int? capacity = extraCapacity.HasValue ? this._states.Length + extraCapacity.Value : null;
-            return new Builder(this, capacity);
+            return new Builder(this, copyPrevious: false);
         }
 
         public static NodeStateTable<T> FromFaultedTable<U>(NodeStateTable<U> table)
         {
             Debug.Assert(table.IsFaulted);
-            return new NodeStateTable<T>(Empty._states, isCompacted: true, table._exception);
-        }
-
-        public static NodeStateTable<T> WithSingleItem(T item, EntryState state)
-        {
-            return new NodeStateTable<T>(ImmutableArray.Create(ImmutableArray.Create((item, state))), isCompacted: false, exception: null);
+            return new NodeStateTable<T>(Empty._states, isCompacted: false, table._exception);
         }
 
         public sealed class Builder
         {
             private readonly ArrayBuilder<ImmutableArray<(T, EntryState)>> _states;
-
+            private readonly NodeStateTable<T> _previous;
             private Exception? _exception = null;
 
-            public Builder(int? capacity = null)
+            internal Builder(NodeStateTable<T> previous, bool copyPrevious = false)
```

Comment:
>= false [](http://example.com/codeflow?start=75&length=7)

Consider removing the default value. It looks like there's only one caller (and that caller passes a value) and it seems helpful to the reader to make this explicit at any call site.

Reply:
Ah yeah, left over from previous.

---
In reply to: [632135635](https://github.com/dotnet/roslyn/pull/53383#discussion_r632135635) [](ancestors = 632135635)